cub/device/dispatch/dispatch_histogram.cuh

```diff
@@ -237,79 +241,56 @@ struct DispatchHistogram
     // Scales samples to evenly-spaced bins
     struct ScaleTransform
     {
+        using CommonT = typename cuda::std::common_type<LevelT, SampleT>::type;
+
+        static constexpr bool PRECOMPUTE_RECIPROCAL =
+          cuda::std::is_floating_point<CommonT>::value;
+
         int    num_bins;    // Number of levels in array
-        LevelT max;         // Max sample level (exclusive)
-        LevelT min;         // Min sample level (inclusive)
-        LevelT scale;       // Bin scaling factor
+        CommonT max;        // Max sample level (exclusive)
+        CommonT min;        // Min sample level (inclusive)
+        CommonT scale;      // Bin scaling factor
 
         // Initializer
-        template <typename _LevelT>
         __host__ __device__ __forceinline__ void Init(
```

Comment:
The `Init` pattern is used in the other helper classes for histogram. This PR is already experiencing quite a bit of scope creep, so I'll leave this as is for consistency with the other helpers. This can be done in a future cleanup.